any_parser/any_parser.py

```diff
@@ -133,6 +138,7 @@ def __init__(self, api_key: str, base_url: str = PUBLIC_SHARED_BASE_URL) -> None
         )
         self._sync_extract_pii = ExtractPIISyncParser(api_key, base_url)
         self._sync_extract_tables = ExtractTablesSyncParser(api_key, base_url)
+        self.batches = BatchParser(api_key, PUBLIC_BATCH_BASE_URL)
```

Comment:
nit: use _batch to indicate this is a private method.

Reply:
Also, can we pass another batch_url and input and pass in the PUBLIC_BATCH_BASE_URL as argument to improve readability.